Cut(eos_designs)!: Remove the default templates from ansible_collections/arista/avd/roles/eos_designs/templates (#5915)
aristanetworks/ansible-avd: docs/porting-guides/6.x.x.md

-   type: l2ls
```

> ### Removal of default Jinja2 templates in eos_designs

Reviewer: which default templates? Please explain more.
Hint, these are the default templates for IP addressing and provide the list of templates.

Author: Updated

-   type: l2ls
```

> ### Removal of default jinja2 templates for `interface_descriptions` and `ip_addressing` in eos_designs

Reviewer 2: ```suggestion
### Removal of default Jinja2 templates for `interface_descriptions` and `ip_addressing` in eos_designs
```

Starting AVD 6.0.0, the default templates for `interface_descriptions` and `ip_addressing` are removed from the eos_designs role. Custom templates may still be provided, but the role no longer includes built-in templates.

The removed default templates for ip_addressing are:

```diff

> - mlg-ibgp-peering-ip-primary.j2

Reviewer 2: Missing `avd-v2-spine-p2p-uplinks-ip.j2` & `avd-v2-spine-p2p-uplinks-peer-ip.j2`?

```suggestion
- avd-v2-spine-p2p-uplinks-ip.j2
- avd-v2-spine-p2p-uplinks-peer-ip.j2
- mlg-ibgp-peering-ip-primary.j2
```

- mlg-ibgp-peering-ip-secondary.j2
- mlag-ip-primary.j2
- mlag-ip-secondary.j2
- mlag-l3-ip-primary.j2
- mlag-l3-ip-secondary.j2
- p2p-uplinks-ip.j2
- p2p-uplinks-peer-ip.j2
- router-id-ipv6.j2
- router-id.j2
- vtep-ip-mlag.j2
- vtep-ip.j2
```

The removed default templates for interface_descriptions are:

```diff
- connected_endpoints/ethernet-interface.j2

> - connected_endpoints/port-channel-interface.j2

Reviewer 2: ```suggestion
- connected_endpoints/ethernet-interfaces.j2
- connected_endpoints/port-channel-interfaces.j2
```

- loopback_interfaces/overlay-loopback.j2
- loopback_interfaces/vtep-loopback.j2
- mlag/ethernet-interface.j2

> - mlag/port-channel-interface.j2

Reviewer 2: ```suggestion
- mlag/ethernet-interfaces.j2
- mlag/port-channel-interfaces.j2
```

- underlay/ethernet-interface.j2

> - underlay/port-channel-interface.j2

Reviewer 2: ```suggestion
- underlay/ethernet-interfaces.j2
- underlay/port-channel-interfaces.j2
```

```

> For migration cases where the legacy templates are required, they can be copied from AVD 5.7 into a custom location and referenced accordingly.

Reviewer 2: It would be very helpful if we could provide an example of where/how exactly to reference path to such copied j2 template for 6.0.0 users that used same default template on 5.x train.

Author: Added

Starting AVD 6.0.0, the default templates for `interface_descriptions` and `ip_addressing` are removed from the eos_designs role. Custom templates may still be provided, but the role no longer includes built-in templates.

The removed default templates for ip_addressing are:

```diff
- avd-v2-spine-p2p-uplinks-ip.j2
- avd-v2-spine-p2p-uplinks-peer-ip.j2
- mlg-ibgp-peering-ip-primary.j2
- mlg-ibgp-peering-ip-secondary.j2
- mlag-ip-primary.j2
- mlag-ip-secondary.j2
- mlag-l3-ip-primary.j2
- mlag-l3-ip-secondary.j2
- p2p-uplinks-ip.j2
- p2p-uplinks-peer-ip.j2
- router-id-ipv6.j2
- router-id.j2
- vtep-ip-mlag.j2
- vtep-ip.j2
```

The removed default templates for interface_descriptions are:

```diff
- connected_endpoints/ethernet-interfaces.j2
- connected_endpoints/port-channel-interfaces.j2
- loopback_interfaces/overlay-loopback.j2
- loopback_interfaces/vtep-loopback.j2
- mlag/ethernet-interfaces.j2
- mlag/port-channel-interfaces.j2
- underlay/ethernet-interfaces.j2
- underlay/port-channel-interfaces.j2
```

For migration cases where the legacy templates are required, they can be copied from AVD 5.7 into a custom location and referenced accordingly.

> For example, after copying the templates to a repository `custom_templates` available in search path when the Ansible playbook is running as [documented here](https://docs.ansible.com/ansible/latest/playbook_guide/playbook_pathing.html):

Reviewer 3: ```suggestion
For migration cases where you must retain legacy logic, copy the required templates from the AVD 5.7 `roles/eos_designs/templates` into a custom template location and reference them explicitly.

As an example, copy the templates to a directory `custom_templates` (make sure it is available in the search paths when the Ansible playbook is executing as [documented here](https://docs.ansible.com/ansible/latest/playbook_guide/playbook_pathing.html)) and update your input variables as follow:
```

> As an example, copy the templates to a directory `custom_templates` (make sure it is available in the search paths when the Ansible playbook is executing as [documented here](https://docs.ansible.com/ansible/latest/playbook_guide/playbook_pathing.html)) and update your input variables as follow:

Author: ```suggestion
As an example, copy the templates to a directory `custom_templates` (make sure it is available in the search paths when the Ansible playbook is executing as [documented here](https://docs.ansible.com/ansible/latest/playbook_guide/playbook_pathing.html)) and update your input variables as follows:
```
